pkg/storage/utils/decomposedfs/spaces.go

```diff
@@ -340,6 +341,37 @@ func (fs *Decomposedfs) UpdateStorageSpace(ctx context.Context, req *provider.Up
 
 // DeleteStorageSpace deletes a storage space
 func (fs *Decomposedfs) DeleteStorageSpace(ctx context.Context, req *provider.DeleteStorageSpaceRequest) error {
+	opaque := req.Opaque
+	var purge bool
+	if opaque != nil {
+		_, purge = opaque.Map["purge"]
+	}
+
+	if purge {
+		ip := fs.lu.InternalPath(req.Id.OpaqueId)
+		matches, err := filepath.Glob(ip)
+		if err != nil {
+			return err
+		}
+
+		// TODO: remove blobs
+		if err := os.RemoveAll(matches[0]); err != nil {
+			return err
+		}
+
+		matches, err = filepath.Glob(filepath.Join(fs.o.Root, "spaces", spaceTypeAny, req.Id.OpaqueId))
+		if err != nil {
+			return err
+		}
+		if len(matches) != 1 {
+			return fmt.Errorf("delete space failed: found %d matching spaces", len(matches))
+		}
+		if err := os.RemoveAll(matches[0]); err != nil {
+			return err
+		}
+		return nil
```

Comment:
NOT CRITICAL. But I just recognized 
```
if err := os.RemoveAll(matches[0]); err != nil { 
    return err
}
return nil 
```
could be written as
```
return os.RemoveAll(matches[0])
```
since it returns in both cases.

Reply:
True, changed.